galsim/config/stamp.py

```diff
@@ -404,61 +387,166 @@ def BuildStamp(config, xsize=0, ysize=0,
+    or config['image'].
+
+    Values given in config['stamp'] take precedence if these are given in both places (which would
+    be confusing, so probably shouldn't do that, but there might be a use case where it would make
+    sense).
+
+    @param config           The configuration dict.
+    @param xsize            The xsize of the image to build (if known).
+    @param ysize            The ysize of the image to build (if known).
+    @param ignore           A list of parameters that are allowed to be in config['stamp']
+                            that we can ignore here.  i.e. it won't be an error if these
+                            parameters are present.
+    @param logger           If given, a logger object to log progress.
+
+    @returns xsize, ysize, image_pos, world_pos
+    """
+    # Check for spurious parameters
+    galsim.config.CheckAllParams(config['stamp'], ignore=ignore)
+
+    # Update the size if necessary
+    if not xsize:
+        if 'xsize' in config['stamp']:
+            xsize = galsim.config.ParseValue(config['stamp'],'xsize',config,int)[0]
+        elif 'size' in config['stamp']:
+            xsize = galsim.config.ParseValue(config['stamp'],'size',config,int)[0]
+        elif 'stamp_xsize' in config['image']:
+            xsize = galsim.config.ParseValue(config['image'],'stamp_xsize',config,int)[0]
+        elif 'stamp_size' in config['image']:
+            xsize = galsim.config.ParseValue(config['image'],'stamp_size',config,int)[0]
+
+    if not ysize:
+        if 'ysize' in config['stamp']:
+            ysize = galsim.config.ParseValue(config['stamp'],'ysize',config,int)[0]
+        elif 'size' in config['stamp']:
+            ysize = galsim.config.ParseValue(config['stamp'],'size',config,int)[0]
+        elif 'stamp_ysize' in config['image']:
+            ysize = galsim.config.ParseValue(config['image'],'stamp_ysize',config,int)[0]
+        elif 'stamp_size' in config['image']:
+            ysize = galsim.config.ParseValue(config['image'],'stamp_size',config,int)[0]
+
+    # Determine where this object is going to go:
+    if 'image_pos' in config['stamp']:
+        image_pos = galsim.config.ParseValue(
+            config['stamp'], 'image_pos', config, galsim.PositionD)[0]
+    elif 'image_pos' in config['image']:
+        image_pos = galsim.config.ParseValue(
+            config['image'], 'image_pos', config, galsim.PositionD)[0]
+    else:
+        image_pos = None
 
-def DrawStamp(psf, gal, config, xsize, ysize, offset, method, logger):
+    if 'world_pos' in config['stamp']:
+        world_pos = galsim.config.ParseValue(
+            config['stamp'], 'world_pos', config, galsim.PositionD)[0]
+    elif 'world_pos' in config['image']:
+        world_pos = galsim.config.ParseValue(
+            config['image'], 'world_pos', config, galsim.PositionD)[0]
+    else:
+        world_pos = None
+
+    return xsize, ysize, image_pos, world_pos
+
+
+def ProfileBasic(config, psf, gsparams, logger):
```

Comment:
Maybe add a verb to this method name?  Without one, it sounds like "profile" *is* the verb, and that implies something very different.